src/main/java/org/isf/medicalstock/manager/MovBrowserManager.java

```diff
@@ -122,4 +151,53 @@ private void check(LocalDateTime from, LocalDateTime to, String errMsgKey) throw
 			}
 		}
 	}
+
+	/**
+	 * Get the last Movement.
+	 *
+	 * @return the retrieved movement.
+	 * @throws OHServiceException
+	 */
+	public Movement getLastMovement() throws OHServiceException {
+		return ioOperations.getLastMovement();
+	}
+
+	/**
+	 * Deletes the last Movement.
+	 *
+	 * @param lastMovement - the last movement to delete
+	 * @throws OHServiceException
+	 */
+	@Transactional
+	public void deleteLastMovement(Movement lastMovement) throws OHServiceException {
+
+		MovementType movType = medicalDsrStockMovTypeManager.getMovementType(lastMovement.getType().getCode());
+		String lotCode = lastMovement.getLot().getCode();
+		Medical medical = lastMovement.getMedical();
+		int medicalCode = medical.getCode();
+		int quantity = lastMovement.getQuantity();
+		if (movType.getType().equals("+")) {
+			medical.setInqty(medical.getInqty() - quantity);
+			medicalsIoOperation.updateMedical(medical);
+			List<Integer> lastMovWithSameLot = ioOperations.getMedicalsFromLot(lotCode);
+			if (lastMovWithSameLot.size() == 1) {
+				lotRepository.deleteById(lotCode);
+			}
+			ioOperations.deleteMovement(lastMovement);
+		} else {
+			Ward ward = lastMovement.getWard();
+			String wardCode = ward.getCode();
+			MedicalWard medWard = movWardBrowserManager.getMedicalWardByWardAndMedical(wardCode, medicalCode, lotCode);
+			medWard.setIn_quantity(medWard.getIn_quantity() - quantity);
+			if (medWard.getIn_quantity() == medWard.getOut_quantity()) {
+				movWardBrowserManager.deleteMedicalWard(medWard);
+			} else {
+				movWardBrowserManager.updateMedicalWard(medWard);
+			}
```

Comment:
I think this is the tricky part: after a discharge `Movement`, the `MedicalWard` could have already been used in the ward itself or to patients, and in such case we cannot delete it anymore. So I think the we need to consider that, like if there are not `MovementWard` for that `MedicalWard` after the `Movement` date, or at least I can't think of anything else at the moment.

After that, if you want to understand if to update the MedicalWard or just delete it (because it was not there before) I think you have to check both `quantity == medWard.getIn_quantity() && medWard.getOut_quantity() == 0`... if so, it means that it has never been moved out from the ward and that the only IN movement was the one we are deleting now.

Comparing `medWard.getIn_quantity() == medWard.getOut_quantity()` doen't make sense to me, but maybe I'm missing some reasoning here.